onyo/main.py

```diff
@@ -232,6 +260,60 @@ def parse_args():
         help='Checks the consistency and validity of the onyo repository and its contents'
     )
     cmd_fsck.set_defaults(run=commands.fsck)
+    # subcommand "set"
+    cmd_set = subcommands.add_parser(
+        'set',
+        help='Set values in assets'
+    )
+    cmd_set.set_defaults(run=commands.set)
+    cmd_set.add_argument(
+        '-n', "--dry-run",
+        required=False,
+        default=False,
+        action='store_true',
+        help='Perform a non-interactive trial run with no changes made'
+    )
+    cmd_set.add_argument(
+        '-q', '--quiet',
+        required=False,
+        default=False,
+        action='store_true',
+        help='Silence output (requires the --yes flag)'
+    )
+    cmd_set.add_argument(
+        '-y', '--yes',
+        required=False,
+        default=False,
+        action='store_true',
+        help='Respond "yes" to any prompts'
+    )
+    cmd_set.add_argument(
+        '-R', '--recursive',
+        required=False,
+        default=False,
+        action='store_true',
+        help='When a directory is given, set values recursively for all assets in it'
```

Comment:
The way this is written is odd to me. If a directory is not provided, but the `-R` is passed, then onyo will ignore it and operate non-recursively?

In general, onoy assumes the top-level dir if no argument is provided, and I think we should be consistent here. If the user says recursive, then onyo will operate recursively.

The following wording is closer to what I'm thinking:

> Set values for all (matching) assets under each directory, recursively,

It's still a bit vague on the "even when no directory is provided" topic, but it feels more clear to me.

Reply:
> The way this is written is odd to me. If a directory is not provided, but the -R is passed, then onyo will ignore it and operate non-recursively?

Yes, when an asset is also given, for example  `onyo set --recursive USB=2 shelf/asset user/`, it will ignore the `--recursive` flag for `shelf/asset` and just run the command recursive for `user/`

Reply:
> In general, onoy assumes the top-level dir if no argument is provided, and I think we should be consistent here. If the user says recursive, then onyo will operate recursively.

Does it assume the top level, or the current working dir?

Reply:
"Set values recursively for assets in directories"

Reply:
> Yes, when an asset is also given, for example onyo set --recursive USB=2 shelf/asset user/, it will ignore the --recursive flag for shelf/asset and just run the command recursive for user/

Well, by definition, it cannot operate recursively on a file, so this makes sense.

But to be clear, if two directories are passed to onyo, both will be operated on recursively. Yes?

But the point I was trying to make is that the wording made it sound like that if we weren't providing any arguments (thus implying the entire repo), then the --recursive flag would no-op. Which I disagree with (and I don't think you intended to say, but it reads that way).